### ISSUE TITLE
Stacks lost after upgrade pulumi from v3.1.0 to v3.2.1

### ISSUE THREAD

**Issue by suraciii** [pulumi/pulumi#6985]
pulumi cannot recognize existing stacks after it upgraded to v3.2.1
``` shell
$ pulumi stack ls
NAME  LAST UPDATE  RESOURCE COUNT

```

## Expected behavior

Stacks should exist after pulumi upgrade to v3.2.1, and all works fine after I downgrade pulumi back to v3.1.0

## Steps to reproduce

1. Create a new stack with local file storage and empty passphrase using pulumi v3.1.0
2. Upgrade pulumi to v3.2.1 using chocolatey
3. pulumi stack ls
4. The stack lost

## Context (Environment)

I tried with pulumi v3.2.0, it has the same issue

**Comment by mikhailshilkov** [comment]
@suraciii Sorry to hear about the trouble.

I'm trying to reproduce this but it works for me both on macOS and Windows... I started with 3.1, created a new aws-ts project with two stacks (test and test2) with local login and empty passphrase, then updated to 3.2.1. `pulumi stack ls` lists both stacks still...

@suraciii Are you able to reproduce it reliably? Or is it one project of yours that went broken?

cc @stack72 could this somehow be related to the recent passphrase changes?

**Comment by lukehoban** [comment]
I believe the state being "lost" is due to https://github.com/pulumi/pulumi/issues/4798. 

The issue affecting blank passphrases was fixed with https://github.com/pulumi/pulumi/pull/6976 and a patch release should be available soon with the fix.

**Comment by mikhailshilkov** [comment]
>  fixed with #6976 and a patch release should be available soon with the fix.

This is already shipped in 3.2.1

**Comment by stack72** [comment]
Hi @suraciii 

Please can you try setting PULUMi_CONFIG_PASSPHRASE="" in your environment and see if "Pulumi stack ls" then works as expected for you?

**Comment by suraciii** [comment]
> Please can you try setting PULUMi_CONFIG_PASSPHRASE="" in your environment and see if "Pulumi stack ls" then works as expected for you?

No, it doesn't work

**Comment by stack72** [comment]
@suraciii 

if you run `pulumi whoami -v` can you check what backend you are logged into? 

I am not able to replace this locally and just wanted to check that you were still logged into your local backend

**Comment by suraciii** [comment]
> if you run pulumi whoami -v can you check what backend you are logged into?

``` shell
$ pulumi whoami -v
User: TANGLEDSHORE\surac
Backend URL: file://.
```

**Comment by dcaba** [comment]
Reproducing this here with `pulumi 3.5.1` and GCS as the backend for the state. `pulumi 3.1.0` works as expected.

**Comment by lukehoban** [comment]
> Reproducing this here with pulumi 3.5.1 and GCS as the backend for the state. pulumi 3.1.0 works as expected.

@dcaba Which issue specifically were you able to reproduce?

> I believe the state being "lost" is due to #4798.

This issue was fixed recently in [3.9.1](https://github.com/pulumi/pulumi/blob/master/CHANGELOG.md#391-2021-07-29).

I am reasonably sure from my understanding of the issues folks were reporting above that those changes will have fixed this, so I'll close it out as a dupe of #4798.  If anyone here is still seeing an issue after upgrading to >=3.9.1, please do share details of what you are seeing now here.

**Comment by dcaba** [comment]
@lukehoban , of course, what I was reproducing was the misbehavior described by this issue. And I confirm pulumi 3.10.2 works again. Thanks!
